Gems/ROS2/Code/Platform/Common/msvc/ros2_static_editor_msvc.cmake

```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+
+set(LY_COMPILE_OPTIONS
+        PRIVATE
+            /EHsc /GR
```

Comment:
O3DE doesn't turn off RTTI, so the `/GR` option is on by [default](https://learn.microsoft.com/en-us/cpp/build/reference/gr-enable-run-time-type-information?view=msvc-170#remarks)